crates/ruff_linter/src/rules/tryceratops/rules/raise_vanilla_class.rs

```diff
@@ -68,7 +68,12 @@ pub(crate) fn raise_vanilla_class(checker: &mut Checker, expr: &Expr) {
     } else {
         expr
     };
-    if checker.semantic().match_builtin_expr(node, "Exception") {
+
+    let semantic = checker.semantic();
+
+    if semantic.match_builtin_expr(node, "Exception")
+        || semantic.match_builtin_expr(node, "BaseException")
```

Comment:
This is resolving the qualified name twice when it only needs to resolve it once -- we can micro-optimize this a little bit:

```suggestion
    if checker
        .semantic()
        .resolve_qualified_name(node)
        .is_some_and(|qualified_name| {
            matches!(qualified_name.segments(), ["" | "builtins", "Exception" | "BaseException"])
        })
```